File: Others/LeetCode/1.Two Sum/two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """    
        hashmap = {}
        for index in range(len(nums)):
            difference = target - nums[index]
            if (difference in hashmap.keys()):
                return [hashmap[difference], index]
            hashmap[nums[index]] = index
        return []


print(Solution.twoSum(Solution,[2,7,11,15], 9))

File: Others/LeetCode/1.Two Sum/two-sum.ts
function twoSum(nums: number[], target: number): number[] {
    let hashmap = new Map();
    for (let i = 0; i < nums.length; i++) {
        const difference = target - nums[i];
        if (Array.from(hashmap.keys()).includes(difference)) {
            return [hashmap.get(difference), i];
        }
        hashmap.set(nums[i], i);
    }
    return [];
};
